### ISSUE TITLE
Refactor DAOs to stateless model

### ISSUE THREAD

**Issue by MikeNeilson** [opendcs/opendcs#796]
**Component to be refactored**
All DAO/DAI classes

**Purpose of refactoring**
OpenDCS 6.8 used stateful DAOs, but assumed the `java.sql.Connection` never changed (e.g. no connection pools, database won't do anything to the database.) OpenDCS 7.0 was the change to a connection pool for CWMS and we kept the stateful idea but made all the DAOs autoclose.

This is still causing major issues:
1. We open too many connections, which slows things down
2. A way around 1 is to put even read operations within a "transaction" and reuse the connection
3. There is a strange interplay with CWMS's use of the Oracle UCP Pool and it's wrapper that's not actually returning the connections properly.
4. Especially within the the SqlDatabaseIO family of classes, all of the connections pool corrections are an impressive, but brittle, hack.

**Proposed method**
Make DAOs no longer stateful. This means they *don't* ever hold a connection.
All DAO methods will take a "connection", actually some form of "access key" (We're open to names) to support various backend types (XML, a SQL Database in combination with a REST API.)

Since the current DAOs *are* stateful, instead of stripping that out in one go, Provide a default method that will retrieve a data "access key", if still created, to perform the operation.

This will provide time to transition all operations to the new scheme.

**Additional context**
Work on this will likely happen after #681, as that *should* reduce the number of places that's required and will reduce additional work to migrate to that concept after all this is done.

**Comment by ktarbet** [comment]
@MikeNeilson  would you please suggest some good starting DAO's to work on for @oskarhurst and myself.
Please check any boxes that don't need work, or are already completed.

### DAO Class Checklist

- [ ] CwmsIntervalDAO  
- [ ] CwmsSiteDAO  
- [ ] CwmsTimeSeriesDAO  
- [ ] CwmsRatingDao  
- [ ] ScreeningDAO  
- [ ] HdbComputationDAO  
- [ ] HdbIntervalDAO  
- [ ] HdbSiteDAO  
- [ ] HdbTimeSeriesDAO  
- [ ] TsdbDao  
- [ ] XmlLoadingAppDAO  
- [ ] XmlPlatformStatusDAO  
- [ ] XmlScheduleEntryDAO  
- [ ] AlarmDAO  
- [ ] AlgorithmDAO  
- [ ] CompDependsDAO  
- [ ] CompDependsNotifyDAO  
- [ ] ComputationDAO  
- [ ] DacqEventDAO  
- [ ] DataTypeDAO  
- [ ] DeviceStatusDAO  
- [ ] EnumSqlDao  
- [ ] LoadingAppDao  
- [ ] PlatformStatusDAO  
- [ ] PropertiesSqlDao  
- [ ] ScheduleEntryDAO  
- [ ] SiteDAO  
- [ ] TsGroupDAO  
- [ ] XmitRecordDAO  
- [ ] OpenTimeSeriesDAO  
- [ ] OpenTsdbIntervalDAO  
- [ ] OpenTsdbXmitRecordDao  
- [ ] EnumXmlDao

**Comment by MikeNeilson** [comment]
Enum, DataType, Alarm, and DacqEvent are probably the easiest to start with.

DacqEvent is probably the easiest to update the usage of as it's only used in a specific logger.

XmitRecord is also only used in a few places.

DataType is small than Alarm.

NOTE: That is a SWAG. I honestly have no idea and it's often the case that things are really nested and intertwined.
